code/datums/mutations/antenna.dm

```diff
@@ -60,9 +60,12 @@
 	action_icon_state = "mindread"
 
 /obj/effect/proc_holder/spell/targeted/mindread/cast(list/targets, mob/living/carbon/human/user = usr)
+	if(!usr.can_cast_magic(MAGIC_RESISTANCE_MIND))
+		return
+
 	for(var/mob/living/M in targets)
-		if(usr.anti_magic_check(FALSE, FALSE, TRUE, 0) || M.anti_magic_check(FALSE, FALSE, TRUE, 0))
-			to_chat(usr, span_warning("As you reach out with your mind, you're suddenly stopped by a vision of a massive tinfoil wall that streches beyond visible range. It seems you've been foiled."))
+		if(M.can_block_magic(MAGIC_RESISTANCE_MIND, charge_cost = 0))
+			to_chat(usr, span_warning("As you reach into [M]'s mind, you are stopped by a mental blockage. It seems you've been foiled."))
```

Comment:
```suggestion
			to_chat(user, span_warning("As you reach into [M]'s mind, you are stopped by a mental blockage. It seems you've been foiled."))
```
Also I can't believe you removed the "wall of tinfoil" joke but kept the punchline ("you've been foiled").

Reply:
This was unfortunate but it had to be done.  `MAGIC_RESISTANCE_MIND` is not exclusive to only tinfoil hats.  Other objects can block that type of magic and in the future others may make more objects that do the same.  Thus I had to remove the tinfoil part indicating the object itself but kept the `It seems you've been foiled.` part because it was neutral.  (and should still be funny if they are wearing a tinfoil hat)